pkg/storage/swift/swift_test.go

```diff
@@ -672,3 +673,29 @@ func TestSwiftIsNotAvailable(t *testing.T) {
 	_, ok := err.(*gophercloud.ErrEndpointNotFound)
 	th.AssertEquals(t, true, ok)
 }
+
+func TestNoPermissions(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+
+	handleAuthentication(t, "object-store")
+
+	th.Mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		th.TestMethod(t, r, "GET")
+		w.Header().Set("Accept-Ranges", "bytes")
+		w.Header().Set("Accept", "application/json")
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+		w.WriteHeader(http.StatusForbidden)
+	})
+
+	d, _ := mockConfig(false, th.Endpoint()+"v3", MockUPISecretNamespaceLister{})
+
+	conn, err := d.getSwiftClient()
+	th.AssertNoErr(t, err)
+
+	// if the user doesn't have permissions, gophercloud should return ErrDefault403
+	listOpts := containers.ListOpts{Full: false}
+	_, err = containers.List(conn, listOpts).AllPages()
+	_, ok := err.(gophercloud.ErrDefault403)
+	th.AssertEquals(t, true, ok)
```

Comment:
Ideally IsSwiftEnabled should be checked there.

Reply:
+1 - we should invoke `IsSwiftEnabled` and expect `false`.

Reply:
done